introduce the board stm32l4p5g_dk disco kit

FRASTM opened the pull request:
Add the STM32L4P5G_DK Discovery kit from STMicroelectronics 
The MCU is the stm32L4P5 from the stm32L4plus family.


Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: arm: new stm32l4p5g disco kit

No HSE on this disco kit, no bypass, use the HSI to source the PLL
Octospi2 is connected to the NOR octoflash through octospim.
like stm32l562, this board cannot read correctly the sdfp
from the NOR octoflash (jedec Id is correctly read).

Signed-off-by: Francois Ramu <francois.ramu@st.com>

louiscr commented:
Wouldn't it also be interesting to add pin definitions for sleep state? More and more drivers switch to pinctrl-1

FRASTM committed:
dts: arm: stm32l4p5Xg devices has 1MB flash

The stm32l4p5xi does not exist yet for the stm32l4 plus serie
Only stm32l4p5Xg with 1MB flash or stm32l4p5Xe with 512KB flash

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Rebase on f7cf252a8cb6cdff0f06e26db1c08fab6f1c2840
Change the dts/st/l4/stm32l4p5 serie with only stm32l4p5Xg or stm32l4p5Xe  (no stm32l4p5Xi available)

FRASTM commented:
> Wouldn't it also be interesting to add pin definitions for sleep state? More and more drivers switch to pinctrl-1

Yes, but I did not see `pinctrl-names = "sleep"; `coming (yet) on any stm32 board
